tests/test_basics.py

```diff
@@ -37,6 +38,31 @@ def error_processor(event, exc_info):
     assert event["exception"]["values"][0]["value"] == "aha! whatever"
 
 
+def test_auto_enabling_integrations_does_not_break(sentry_init, caplog):
+    caplog.set_level(logging.DEBUG)
+
+    sentry_init(_experiments={"auto_enabling_integrations": True}, debug=True)
```

Comment:
Since integrations are enabled only once globally, isn't [`sentry_init`](https://github.com/getsentry/sentry-python/blob/dd85044c0ec76b19bc7f5b2984c3d4dcc47567ed/tests/conftest.py#L182-L197) influenced by the other tests?

Reply:
The monkeypatches themselves leak, but not integration settings or which integrations are really enabled for the current client. The code we patch in should not execute unless the integration is marked as enabled on the current hub.

Some tests we mark with `pytest.mark.forked` to execute them in the subprocess as that's the most surefire way to make sure tests do not make other tests fail. This affects test times negatively so we use it sparingly. There's a CLI option too if you need to debug a state leak.